ethereum-consensus/src/electra/epoch_processing.rs

```diff
@@ -0,0 +1,185 @@
+use crate::electra::{
+    compute_activation_exit_epoch, decrease_balance, get_current_epoch,
+    helpers::{
+        get_activation_exit_churn_limit, get_active_balance, initiate_validator_exit,
+        is_eligible_for_activation_queue, switch_to_compounding_validator,
+    },
+    increase_balance, is_active_validator, is_eligible_for_activation, BeaconState, Context, Error,
+};
+
+pub fn process_registry_updates<
+    const SLOTS_PER_HISTORICAL_ROOT: usize,
+    const HISTORICAL_ROOTS_LIMIT: usize,
+    const ETH1_DATA_VOTES_BOUND: usize,
+    const VALIDATOR_REGISTRY_LIMIT: usize,
+    const EPOCHS_PER_HISTORICAL_VECTOR: usize,
+    const EPOCHS_PER_SLASHINGS_VECTOR: usize,
+    const MAX_VALIDATORS_PER_COMMITTEE: usize,
+    const SYNC_COMMITTEE_SIZE: usize,
+    const BYTES_PER_LOGS_BLOOM: usize,
+    const MAX_EXTRA_DATA_BYTES: usize,
+    const PENDING_BALANCE_DEPOSITS_LIMIT: usize,
+    const PENDING_PARTIAL_WITHDRAWALS_LIMIT: usize,
+    const PENDING_CONSOLIDATIONS_LIMIT: usize,
+>(
+    state: &mut BeaconState<
+        SLOTS_PER_HISTORICAL_ROOT,
+        HISTORICAL_ROOTS_LIMIT,
+        ETH1_DATA_VOTES_BOUND,
+        VALIDATOR_REGISTRY_LIMIT,
+        EPOCHS_PER_HISTORICAL_VECTOR,
+        EPOCHS_PER_SLASHINGS_VECTOR,
+        MAX_VALIDATORS_PER_COMMITTEE,
+        SYNC_COMMITTEE_SIZE,
+        BYTES_PER_LOGS_BLOOM,
+        MAX_EXTRA_DATA_BYTES,
+        PENDING_BALANCE_DEPOSITS_LIMIT,
+        PENDING_PARTIAL_WITHDRAWALS_LIMIT,
+        PENDING_CONSOLIDATIONS_LIMIT,
+    >,
+    context: &Context,
+) -> Result<(), Error> {
+    let current_epoch = get_current_epoch(state, context);
+    for i in 0..state.validators.len() {
+        let validator = &mut state.validators[i];
+        if is_eligible_for_activation_queue(validator, context) {
+            validator.activation_eligibility_epoch = current_epoch + 1;
+        }
+        if is_active_validator(validator, current_epoch) &&
+            validator.effective_balance <= context.ejection_balance
+        {
+            initiate_validator_exit(state, i, context)?;
+        }
+    }
+
+    let activation_epoch = compute_activation_exit_epoch(current_epoch, context);
+    for i in 0..state.validators.len() {
+        let validator = &state.validators[i];
+        if is_eligible_for_activation(state, validator) {
+            let validator = &mut state.validators[i];
+            validator.activation_epoch = activation_epoch;
```

Comment:
makes the line a bit more dense but you don't quite have the stutter
```suggestion
            state.validators[i].activation_epoch = activation_epoch;
```